### PR TITLE
Enhancement/25 unit tests part 2

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        this.sqlFunctionScalarCast = new SqlFunctionScalarCast(this.dataType, this.arguments);
+    }
+
+    @Test
+    void testGetArguments() {
+        assertThat(this.sqlFunctionScalarCast.getArguments(), equalTo(this.arguments));
+    }
+
+    @Test
+    void testGetArgumentsEmptyListThrowsException() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java]: Test name seems off.
Maybe `testGetArgumentsWithNullAsArgumentList`?

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        assertThat(this.sqlFunctionScalarCast.getType(), equalTo(SqlNodeType.FUNCTION_SCALAR_CAST));
+    }
+
+    @Test
+    void testGetFunction() {
+        assertThat(this.sqlFunctionScalarCast.getFunction(), equalTo(ScalarFunction.CAST));
+    }
+
+    @Test
+    void testAccept() throws AdapterException {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java]: Now that I've seen this method being repeated almost identically a number of times I think we should extract the common parts. Generics should do the trick.
`assertAcceptVisitor(SqlNodeVisitor<? extends SqlNode> visitor)`

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarExtractTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        assertThat(this.sqlFunctionScalarExtract.getToExtract(), equalTo(TEST_STRING_TO_EXTRACT));
+    }
+
+    @Test
+    void testGetArguments() {
+        assertThat(this.sqlFunctionScalarExtract.getArguments(), equalTo(this.arguments));
+    }
+
+    @Test
+    void testGetArgumentsEmptyListThrowsException() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlFunctionScalarExtractTest.java]: See above.

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java (hunk 2 above)
Comment (redcatbear) [reply src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java]: Theoretically we could additionally use parameterized tests here where the list of classes is the parameter list. But that would distribute the tests for each class over two test classes and that is harder to maintain.

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarExtractTest.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlFunctionScalarExtractTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCastTest.java]: It's impossible,  because visit() is overloaded method

**Pull request merged**